workshops/views.py

```diff
@@ -156,6 +157,22 @@ def get_success_message(self, cleaned_data):
         return self.success_message.format(cleaned_data, name=str(self.object))
 
 
+class DeleteViewContext(DeleteView):
+    """
+    Class-based view for deleting objects that extends default template context
+    by adding proper page title.
+    """
+    success_message = '{} was deleted successfully.'
+
+    def delete(self, request, *args, **kwargs):
+        '''Workaround for https://code.djangoproject.com/ticket/21926'''
+        messages.success(
+            self.request,
+            self.success_message.format(self.get_object())
+        )
+        return super().delete(request, *args, **kwargs)
+
+
```

Comment:
Yes, we currently also redirect right after deletion. There is no intermediate confirmation page (except for the confirm javascript popup).